src/Symfony/Bridge/Doctrine/CheckConnection.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Exception as DBALException;
+
+class CheckConnection
+{
+    public static function checkConnection(Connection $connection)
+    {
+        try {
+            $connection->executeQuery($connection->getDatabasePlatform()->getDummySelectSQL());
+        } catch (DBALException) {
+            $connection->close();
+            // Attempt to reestablish the lazy connection by sending another query.
+            $connection->executeQuery($connection->getDatabasePlatform()->getDummySelectSQL());
```

Comment:
It needs to look like this to be compatible with DBAL 4
```suggestion
            $connection->executeQuery($connection->getDatabasePlatform($connection)->getDummySelectSQL());
        } catch (DBALException) {
            $connection->close();
            // Attempt to reestablish the lazy connection by sending another query.
            $connection->executeQuery($connection->getDatabasePlatform($connection)->getDummySelectSQL());
```